golem-component-service/db/migration/postgres/005__add_plugins.sql

```diff
@@ -0,0 +1,32 @@
+CREATE TABLE plugins
+(
+    name                 text     NOT NULL,
+    version              text     NOT NULL,
+    description          text     NOT NULL,
+    icon                 bytea    NOT NULL,
+    homepage             text     NOT NULL,
+    plugin_type          smallint NOT NULL,
+    scope_component_id   uuid REFERENCES components (component_id),
+    provided_wit_package text,
+    json_schema          text,
+    validate_url         text,
+    transform_url        text,
+    component_id         uuid REFERENCES components (component_id),
+    component_version    bigint,
+    deleted              boolean  NOT NULL DEFAULT FALSE,
```

Comment:
it seems that we do need to use this in almost every query, so i would add a covering index for the primary key columns and this together